components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -4505,6 +4511,45 @@ private void updatePolicyGroups(int applicationId, Object[] policyPartialList,
 		}
 	}
 
+	/**
+	 * Update Java policies
+	 *
+	 * @param applicationId
+	 * @param javaPolicyList
+	 * @param connection
+	 * @throws java.sql.SQLException on error
+	 */
+	private void updateJavaPolicies(int applicationId, Object[] javaPolicyList,
+									Connection connection) throws
+			SQLException {
+
+		PreparedStatement ps = null;
+		String query = "DELETE FROM APM_APP_JAVA_POLICY_MAPPING WHERE APP_ID=? ";
+
+		try {
+			if (log.isDebugEnabled()) {
+				log.debug("Updating Java Policies mappings with webapp id : " + applicationId);
+			}
+			ps = connection.prepareStatement(query);
+			ps.setInt(1, applicationId);
+			ps.executeUpdate();
+
+			//Save application's policy partial mapping data
+			saveJavaPolicyMappings(connection, applicationId, javaPolicyList);
+
+		} catch (SQLException e) {
+			/*
+			In the code im using a single SQL connection passed from the parent function so I'm logging the error here
+			and throwing the SQLException so the connection will be disposed by the parent function.
+			*/
+			log.error("Error while deleting XACML policy partial mappings for webapp : " +
+					applicationId, e);
+			throw e;
```

Comment:
This function is a private function and the SQL exception is thrown within the DAO layer